tests/unit/amo/components/TestFeaturedAddonReview.js

```diff
@@ -0,0 +1,182 @@
+import * as React from 'react';
+
+import { fetchReview, setReview } from 'amo/actions/reviews';
+import FeaturedAddonReview, {
+  FeaturedAddonReviewBase,
+} from 'amo/components/FeaturedAddonReview';
+import AddonReviewCard from 'amo/components/AddonReviewCard';
+import { createApiError } from 'core/api';
+import { ErrorHandler } from 'core/errorHandler';
+import { dispatchClientMetadata, fakeReview } from 'tests/unit/amo/helpers';
+import {
+  createStubErrorHandler,
+  fakeI18n,
+  createFakeLocation,
+  shallowUntilTarget,
+  createContextWithFakeRouter,
+  createFakeHistory,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const getProps = ({
+    location = createFakeLocation(),
+    params,
+    ...customProps
+  } = {}) => {
+    return {
+      errorHandler: createStubErrorHandler(),
+      i18n: fakeI18n(),
+      location,
+      store,
+      ...customProps,
+    };
+  };
+
+  const render = ({ ...customProps } = {}) => {
+    const props = getProps(customProps);
+
+    return shallowUntilTarget(
+      <FeaturedAddonReview {...props} />,
+      FeaturedAddonReviewBase,
+      {
+        shallowOptions: createContextWithFakeRouter({
+          history: createFakeHistory(),
+        }),
+      },
+    );
+  };
+
+  it('fetches a review on mount', () => {
+    const reviewId = 1;
+    const dispatch = sinon.stub(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    render({
+      errorHandler,
+      reviewId,
+    });
+
+    sinon.assert.calledWith(
+      dispatch,
+      fetchReview({
+        reviewId,
+        errorHandlerId: errorHandler.id,
+      }),
+    );
+  });
+
+  it('fetches a review when the reviewId changes', () => {
+    const firstReviewId = 1;
+    const secondReviewId = 1;
```

Comment:
It would be helpful to change the ID here 😎 

Actually, I think this test is also missing some setup, right?

```js
    store.dispatch(setReview({ ...fakeReview, id: firstReviewId }));
```

Reply:
It's great to get into the habit of watching your test fail at least once. You can do it by writing the tests first or commenting out the code that passes the test.

Reply:
Oops! Yes, I usually do try to make sure I can make a test fail, but I guess I failed (pun intended) at that this time.

As for the setup, we don't strictly need that, as it's okay for the component to fetch the review for the first id and then also fetch the review for the second id. That's why I'm using `dispatch.resetHistory()`. I think the test is still valid, although we are asking the component to fetch two reviews. Do you feel it is important to change it and make sure the first review is already loaded during test setup?

Reply:
> Do you feel it is important to change it and make sure the first review is already loaded during test setup?

No, that's not important if you don't think it's necessary. I haven't looked at the newer patch but I couldn't see how the old patch was successfully handling changes from one review to another. That logic *is* important. I'll look at the new patch.